packages/application-shell/src/components/setup-flop-flip-provider/setup-flop-flip-provider.tsx

```diff
@@ -17,6 +17,40 @@ import type {
 } from '../../types/generated/mc';
 import AllFeaturesQuery from './fetch-all-features.mc.graphql';
 
+const WAIT_FOR_REMOTE_FLAGS_TIMEOUT = 500; // milliseconds
+
+type TStatusAwareRendererProps = {
+  shouldWaitForRemoteFlags?: boolean;
+  children: ReactNode;
+};
+const StatusAwareRenderer = (props: TStatusAwareRendererProps) => {
+  const [shouldRender, setShouldRender] = useState(
+    !props.shouldWaitForRemoteFlags
+  );
+  const { isReady } = useAdapterStatus();
+  const displayTimeout = useRef<NodeJS.Timeout>();
+
+  if (isReady && !shouldRender) {
+    setShouldRender(true);
+    clearTimeout(displayTimeout.current);
+  }
+
+  useEffect(() => {
+    displayTimeout.current = setTimeout(() => {
+      setShouldRender(true);
+    }, WAIT_FOR_REMOTE_FLAGS_TIMEOUT);
+
+    return () => {
+      clearTimeout(displayTimeout.current);
+    };
+  }, []);
+
+  return shouldRender ? <>{props.children}</> : null;
+};
+StatusAwareRenderer.defaultProps = {
+  shouldWaitForRemoteFlags: process.env.NODE_ENV === 'test' ? false : true,
```

Comment:
By default we will delay the rendering so consumer applications don't have to be updated.